Update RELEASE_NOTES for v5.3.0

andreas-bok-sociomantic opened the pull request:
The RELEASE_NOTES might be subject to additions depending on how we decide to 
structure the v5.2.x point releases. 

Thus it is labeled as `status-blocked` so far

andreas-bok-sociomantic committed:
Update RELEASE_NOTES for v5.3.0

[ci skip]

andreas-bok-sociomantic commented:
Additions to Milestone v5.2.x is freezed.

ben-palmer-sociomantic commented on RELEASE_NOTES.md:
```diff
@@ -8,6 +8,42 @@ autoconf   | >=2.61
 libtool    | >=2.x
 
-New Features
+This release integrates changes which were merged upstream, but after the **Felmingham v5.1.4**
```
Should we have this kind of release "summary" at the start of the release notes? Where it is at the moment it looks like it is in the dependencies.

andreas-bok-sociomantic replied:
I agree, it looks a bit strange. 
I could move it to the relevant place in the list of **Issues Fixed**.

ben-palmer-sociomantic commented on RELEASE_NOTES.md:
```diff
@@ -8,6 +8,42 @@ autoconf   | >=2.61
+
+* Change location of generated documentation ([#37](https://github.com/sociomantic-tsunami/libdrizzle-redux/pull/37))
+
+* Minor fixes and improvements ([#28](https://github.com/sociomantic-tsunami/libdrizzle-redux/pull/28), [#31](https://github.com/sociomantic-tsunami/libdrizzle-redux/pull/31), [#32](https://github.com/sociomantic-tsunami/libdrizzle-redux/pull/32), [#38](https://github.com/sociomantic-tsunami/libdrizzle-redux/pull/38), [#44](https://github.com/sociomantic-tsunami/libdrizzle-redux/pull/44), [#49](h ...
```
These are in order except for 65 and 68.

andreas-bok-sociomantic commented:
Updated based on comments from Ben. 
- fix incorrect numerical order
- reformat content for better legibility 
- split up the summary and moved it to the relevant item in the list of `issues fixed`

ben-palmer-sociomantic commented on RELEASE_NOTES.md:
```diff
@@ -8,6 +8,55 @@ autoconf   | >=2.61
+  `non_blocking` option being set to true.
+	([#41](https://github.com/sociomantic-tsunami/libdrizzle-redux/pull/41))
+
+* Optimization of `drizzle_field_buffer()` so an array of `drizzle_field_t`
```
Is this actually a new feature?

andreas-bok-sociomantic replied:
No. It was the reason for having a summary at the top. 
Looked at different ways of writing the RELEASE_NOTES, and didn't find a definite guideline. 
Wanted to give a description of the central changes in the release, but patch releases are normally only a list of things fixed, it doesn't seem to fit the bil. 
Was considering changing it to a minor release, since the async binlog is sort of a new feature.

ben-palmer-sociomantic commented on RELEASE_NOTES.md:
```diff
@@ -8,6 +8,55 @@ autoconf   | >=2.61
+	([#34](https://github.com/sociomantic-tsunami/libdrizzle-redux/pull/34),
+	 [#53](https://github.com/sociomantic-tsunami/libdrizzle-redux/pull/53))
+
+* Add continuous integration testing for clang compiler
```
Same here

ben-palmer-sociomantic commented on RELEASE_NOTES.md:
```diff
@@ -8,6 +8,55 @@ autoconf   | >=2.61
+* Fixed integer underflow when processing large result sets using prepared statements
+	([#47](https://github.com/sociomantic-tsunami/libdrizzle-redux/pull/47))
+
+* Add unittest for event_watch_fn callback function
```
Same here

ben-palmer-sociomantic commented:
I think a few of the bug fixes might actually be new features.

andreas-bok-sociomantic reviewed (commented):
Ah ok, so maybe add a section with `New Features`, on the other hand it goes against the semantics of a point release .

ben-palmer-sociomantic commented:
> Ah ok, so maybe add a section with New Features, on the other hand it goes against the semantics of a point release .

> Was considering changing it to a minor release, since the async binlog is sort of a new feature.

Do you have a link to the guideline somewhere?

andreas-bok-sociomantic commented:
> Do you have a link to the guideline somewhere? 
No it was compiled from looking at different projects, ours and external. 
E.g. https://github.com/sociomantic-tsunami/git-hub/releases uses [summary/new features/issues fixed] sections for minor/major releases and only [issues fix or changelog] section for point releases.
Imagemagick has no info ... https://github.com/ImageMagick/ImageMagick/releases

andreas-bok-sociomantic commented:
Deciding some of the fixes are new (non-breaking) features, the http://semver.org wuold suggest minor release, and hence the sections: **summary/new features/issues fixed**

andreas-bok-sociomantic commented:
Rewrote RELEASE NOTES as minor release

ben-palmer-sociomantic commented on RELEASE_NOTES.md:
```diff
@@ -1,5 +1,20 @@
+
+These notes are usually structured in 4 sections: **Migration Instructions**,
+which are the mandatory steps a user must do to update to the new version,
+**Deprecated**, which contains deprecated functions which are not recommended to
```
Should this be `Deprecations`?

andreas-bok-sociomantic replied:
Yes

ben-palmer-sociomantic commented on RELEASE_NOTES.md:
```diff
@@ -1,5 +1,20 @@
+incrementally, following the instructions in the previous versions' release
+notes.
+
+These notes are usually structured in 4 sections: **Migration Instructions**,
```
What about the `Dependencies` section?

andreas-bok-sociomantic replied:
It is empty for this release. Guess it makes more sense to exclude it

ben-palmer-sociomantic replied:
Is it empty? We haven't added anything to it but its still there right?

leandro-lucarella-sociomantic replied:
Was this resolved? Dependencies should stay as long as this project depends on anything else :)

andreas-bok-sociomantic replied:
Nope, I will add back the section.

andreas-bok-sociomantic commented:
Updated 
```diff

andreas-bok-sociomantic commented:
@leandro-lucarella-sociomantic, any comments on the structuring/phrasing of the **RELEASE_NOTES**?

leandro-lucarella-sociomantic commented on RELEASE_NOTES.md:
```diff
@@ -8,6 +24,64 @@ autoconf   | >=2.61
+
+* The generated documentation is now written to a top-level folder `sphinx-build`.
+  Since `clean*` targets have been changed accordingly, documentation generated
+  prior to this release have to be removed manually.
```
This is just a build artefact, it doesn't really affect users, right?

andreas-bok-sociomantic replied:
> his is just a build artefact, it doesn't really affect users, right?

Yes 
Agree, putting it in **Migration Instructions** is a bit misleading. Will put it under **Issues Fixed**

leandro-lucarella-sociomantic reviewed (commented):
Besides the tiny comments, LGTM.

andreas-bok-sociomantic commented:
Updated based on comments from @leandro-lucarella-sociomantic 
```diff 
diff --git a/RELEASE_NOTES.md b/RELEASE_NOTES.md
index 7f56c5f..d436e77 100644
--- a/RELEASE_NOTES.md
+++ b/RELEASE_NOTES.md
@@ -12,7 +12,6 @@ be used (and will be removed in the next major release) but will not break any
 old code, **New Features** which are new features available in the new
 version that users might find interesting; **Issues Fixed** which contains
 bugfixes and improvements that doesn't extend functionality.
-Empty sections are omitted.
 
 Dependencies
 ------------
@@ -27,9 +26,8 @@ libtool    | >=2.x
 Migration Instructions
 ----------------------
 
-* The generated documentation is now written to a top-level folder `sphinx-build`.
-  Since `clean*` targets have been changed accordingly, documentation generated
-  prior to this release have to be removed manually.
+Deprecations
+------------
 
 New Features
 ------------
@@ -43,13 +41,13 @@ New Features
 Issues Fixed
 ------------
 
-* update hardcoded repo location
+* Update hardcoded repo location
     ([#28](https://github.com/sociomantic-tsunami/libdrizzle-redux/pull28))
 * Fix version number in configure script
     ([#31](https://github.com/sociomantic-tsunami/libdrizzle-redux/pull31))
 * Update readme and doc
     ([#32](https://github.com/sociomantic-tsunami/libdrizzle-redux/pull32))
-* Change location of generated documentation
+* Change location of generated documentation to top-level folder `sphinx-build`
     ([#37](https://github.com/sociomantic-tsunami/libdrizzle-redux/pull/37))
 * Restructure man page doc generation
     ([#38](https://github.com/sociomantic-tsunami/libdrizzle-redux/pull38))
```

leandro-lucarella-sociomantic commented on RELEASE_NOTES.md:
```diff
@@ -8,6 +23,63 @@ autoconf   | >=2.61
+
+Deprecations
+------------
+
```
I wouldn't put empty sections, they are just confusing.

andreas-bok-sociomantic replied:
>  Dependencies should stay as long as this project depends on anything else :)

Misread **Dependencies** as **Deprecations** ...

andreas-bok-sociomantic commented:
Updated 
```diff
diff --git a/RELEASE_NOTES.md b/RELEASE_NOTES.md
index d436e77..3094d20 100644
--- a/RELEASE_NOTES.md
+++ b/RELEASE_NOTES.md
@@ -23,12 +23,6 @@ autoconf   | >=2.61
 zlib1g-dev | >=0.13.x
 libtool    | >=2.x
 
-Migration Instructions
-----------------------
-
-Deprecations
-------------
-
 New Features
 ------------
```

leandro-lucarella-sociomantic reviewed (approved):
LGTM

andreas-bok-sociomantic commented:
Updated, added PR to list of issues fixed

andreas-bok-sociomantic commented:
do you have more comments @ben-palmer-sociomantic?

ben-palmer-sociomantic commented on RELEASE_NOTES.md:
```diff
@@ -1,5 +1,20 @@
+be used (and will be removed in the next major release) but will not break any
+old code, **New Features** which are new features available in the new
+version that users might find interesting; **Issues Fixed** which contains
+bugfixes and improvements that doesn't extend functionality.
```
Should be `that don't extend functionality`.

andreas-bok-sociomantic replied:
Indeed

ben-palmer-sociomantic commented:
One minor comment on the grammar, otherwise LGTM.

andreas-bok-sociomantic commented:
Fixed grammatical error

andreas-bok-sociomantic commented:
I will close this PR without merging. 
`libdrizzle v5.3.0` is going into labor.
